paramiko/transport.py

```diff
@@ -406,6 +406,7 @@ def __init__(
                     # addr = sockaddr
                     sock = socket.socket(af, socket.SOCK_STREAM)
                     try:
+                        sock.settimeout(self._active_check_timeout)
```

Comment:
`_active_check_timeout` is 0.1 seconds, seems not quite enough for a reasonable connect() timeout

I'd use `banner_timeout` but that's set-up later and can't be configured by the user until after `Transport.__init__()` completes. Maybe best to just use the default `banner_timeout` value of 15 seconds here ...

Reply:
Thanks! I'll give it a try.